build.gradle

```diff
@@ -1053,7 +1053,51 @@ distTar.mustRunAfter clean, cleanAll
 distZip.mustRunAfter clean, cleanAll
 distProduct.mustRunAfter clean, cleanAll
 
+
+
+task deleteDocsDir(type: Delete) {
+  delete "${rootProject.buildDir}/docs"
+}
+
+task docs(type: ScalaDoc) {
+  apply plugin: 'scala'
+  dependsOn deleteDocsDir
+  Set<String> allSource = []
+  def docProjects = rootProject.subprojects.collectMany { project ->
```

Comment:
Its a problem with gradle configuration. Looks like if not needed gradle does not configure  or initializes  a project. Hence hasPlugin('scala') fails and all scala files are skipped.